src/query/pipeline/sources/src/processors/sources/input_formats/input_pipeline.rs

```diff
@@ -203,13 +204,28 @@ pub trait InputFormatPipe: Sized + Send + 'static {
                 let splits = splits.to_owned().clone();
                 tokio::spawn(async move {
                     let mut futs = FuturesUnordered::new();
-                    for s in &splits {
-                        let fut = Self::read_split(ctx_clone2.clone(), s);
+                    for s in splits.into_iter() {
```

Comment:
To run batch futures, we can try this way?
https://github.com/datafuselabs/databend/blob/757415a8bb741fed56e9b6307122834d6f8766f3/src/query/storages/fuse/src/io/snapshots.rs#L93-L117

Reply:
here we need to process a future result as soon as it finishes in stead of try_join_all.

Reply:
need to wrap a future of  `read_split + send` to use try_spawn_batch.
using semaphore if fully pipelined compared with `splits.chunks(p)`  in which batches is not ovelaped

Reply:
since this  pr is a bug fix,  can I do the refactor in a separate pr later?